src/SprykerEco/Zed/PunchoutCatalogs/Communication/Form/ConnectionSetupSubForms/PunchoutCatalogConnectionSetupForm.php

```diff
@@ -0,0 +1,352 @@
+            ->addCompanyUserField($builder, [])
+            ->addCompanyUserFieldListeners($builder);
+    }
+
+    /**
+     * @param \Symfony\Component\OptionsResolver\OptionsResolver $resolver
+     *
+     * @return void
+     */
+    public function configureOptions(OptionsResolver $resolver): void
+    {
+        $resolver->setDefaults([
+            'data_class' => PunchoutCatalogConnectionSetupTransfer::class,
+        ]);
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addLoginModeField(FormBuilderInterface $builder)
+    {
+        $builder->add(PunchoutCatalogConnectionSetupTransfer::LOGIN_MODE, ChoiceType::class, [
+            'label' => static::FIELD_LABEL_LOGIN_MODE,
+            'choices' => [
+                'single_user' => static::LOGIN_MODE_SINGLE_USER,
+                'dynamic_user_creation' => static::LOGIN_MODE_DYNAMIC_USER,
+            ],
+            'attr' => [
+                'class' => 'dependent-trigger',
+                'data-dependent-group' => static::DEPENDENT_GROUP_LOGIN_MODE,
+            ],
+            'constraints' => new NotBlank(),
+        ]);
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return $this
+     */
+    protected function addCompanyBusinessUnitField(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add(PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_BUSINESS_UNIT, SelectType::class, array_merge(
+            $this->getCompanyBusinessUnitFieldOptions(),
+            $options
+        ));
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     * @param array $options
+     *
+     * @return $this
+     */
+    protected function addCompanyUserField(FormBuilderInterface $builder, array $options)
+    {
+        $builder->add(
+            PunchoutCatalogConnectionSetupTransfer::FK_COMPANY_USER,
+            SelectType::class,
+            array_merge($this->getCompanyUserFieldOptions(), $options)
+        );
+
+        return $this;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormBuilderInterface $builder
+     *
+     * @return $this
+     */
+    protected function addCompanyBusinessUnitFieldListeners(FormBuilderInterface $builder)
+    {
+        $formModificationCallback = function (FormEvent $event) {
+            $form = $event->getForm();
+            $loginMode = $event->getData()[PunchoutCatalogConnectionSetupTransfer::LOGIN_MODE] ?? null;
+
+            if ($loginMode !== static::LOGIN_MODE_DYNAMIC_USER) {
+                $this->disableCompanyBusinessUnitField($form);
+
+                return;
+            }
+
+            $idParentCompanyBusinessUnit = $form
+                ->getParent()
+                ->getParent()
```

Comment:
`getParent()->getParent()` already a bad sign. Why do you need this?
Is there any other way of achieving this?

Reply:
In order to access main form data.
ATM I see only 1 way on how to get rid of this:
- Have different form structure

But since it will require significant changes, so I would leave this till TL review, if you don't mind.